src/any_agent/evaluation/agent.py

```diff
@@ -1,121 +1,85 @@
-import json
-
-from pydantic import BaseModel
+from collections.abc import Callable
+from typing import Any
 
 from any_agent import AgentConfig, AnyAgent
+from any_agent.config import AgentFramework
 from any_agent.evaluation.schemas import AgentOutput
+from any_agent.evaluation.tools import AgentEvaluationTools
 from any_agent.tracing.agent_trace import AgentTrace
+from any_agent.utils.asyncio_sync import run_async_in_sync
 
-MAX_EVIDENCE_LENGTH: int = 500
-
+AGENT_INSTRUCTIONS = f"""You are a helpful assistant that will be used to evaluate the correctness of an agent trace. Given a specific question regarding the quality of the something about the agent, utilize the appropriate tools in order to gather the answer needed in order to accurately answer the question. If you are asked anything about what the agent did, you should strongly consider using the get_evidence_from_spans tool to get the evidence. However, if the question is about specific details of the agent's actions, you don't necessarily need to use the get_evidence_from_spans tool.
 
-class AgentTooling:
-    def __init__(self, trace: AgentTrace):
-        self.trace = trace
+Answer with:
+1. "passed": true or false
+2. "reasoning": Brief explanation for your decision
 
-    def get_final_output(self) -> str | BaseModel | None:
-        """Get the final output from the agent trace.
+Your final answer must be a valid JSON string that conforms to the following JSON schema:
 
-        Returns:
-            str | BaseModel | None: The final output of the agent
+{AgentOutput.model_json_schema()}
+"""
 
-        """
-        return self.trace.final_output
 
-    def get_tokens_used(self) -> int:
-        """Get the number of tokens used by the agent as reported by the trace.
+class AgentAsJudge:
```

Comment:
Note the same issue with using a class, as related below. The class attributes will not last, or be modified, across invocations. A higher level function or just adding the current attributes as params is imho again simpler and cleaner.

Reply:
Just to clarify, I'd use a curried or higher order function for this instead of classes, but as far as it works, fine for me.